receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/stats_provider.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecscontainermetrics
+
+import (
+	"encoding/json"
+)
+
+// StatsProvider wraps a RestClient, returning an unmarshaled metadata and docker stats
+type StatsProvider struct {
+	rc RestClient
+}
+
+// NewStatsProvider returns a new stats provider
+func NewStatsProvider(rc RestClient) *StatsProvider {
+	return &StatsProvider{rc: rc}
+}
+
+// GetStats calls the ecs task metadata endpoint and unmarshals the data
+func (p *StatsProvider) GetStats() (map[string]ContainerStats, TaskMetadata, error) {
```

Comment:
I think in #1044 we agreed that container metadata needs to be in the receiver, but task metadata I still think we want in a processor that adds to `Resource`, so it can be used for this, traces, and other metrics. Was it not the case that for task attributes that would be ok? Currently we're not planning on adding these to every SDK since we get more bang-for-buck with a collector-first implementation (this was just decided today though)

Reply:
Hi @anuraaga , I believe that discussion was more related to setting `labels/dimensions/metadata` to task level metrics, and later we agreed to add them in the receiver for now. We also know that, if you implement a process later, we will do  some redundant works. However, regardless of the label settings, I need to parse this `TaskMetadata` for calculating some task level metrics value. We cannot skip it. For example, for calculating the value for `MemoryReserved`, first we check the task level memory which comes from this `TaskMetadata`. If that is not defined, we get the some for all containers.